pystoned/CNLS.py

```diff
@@ -223,46 +219,38 @@ def afriat_rule(model, i, h):
 
                 return afriat_rule
 
-        # TODO(error handling): replace with undefined model attribute
-        return False
+        raise ValueError("Undefined model parameters")
 
     def display_status(self):
         """Display the status of problem"""
         if self.optimization_status == 0:
-            print("Model isn't optimized. Use optimize() method to estimate the model.")
-            return False
+            raise Exception("Model isn't optimized. Use optimize() method to estimate the model.")
         print(self.display_status)
 
     def display_alpha(self):
         """Display alpha value"""
         if self.optimization_status == 0:
-            print("Model isn't optimized. Use optimize() method to estimate the model.")
-            return False
+            raise Exception("Model isn't optimized. Use optimize() method to estimate the model.")
         self.__model__.alpha.display()
 
     def display_beta(self):
         """Display beta value"""
         if self.optimization_status == 0:
-            print("Model isn't optimized. Use optimize() method to estimate the model.")
-            return False
+            raise Exception("Model isn't optimized. Use optimize() method to estimate the model.")
         self.__model__.beta.display()
 
     def display_lamda(self):
         """Display lamda value"""
         if self.optimization_status == 0:
-            print("Model isn't optimized. Use optimize() method to estimate the model.")
-            return False
+            raise Exception("Model isn't optimized. Use optimize() method to estimate the model.")
         if type(self.z) == type(None):
-            # TODO: Replace print by warning
-            print("Without z variable")
-            return
+            raise Exception("Cannot retrieve z variables, since the model is built without z variables.")

```

Comment:
should be:

` raise Exception("Estimated coefficient (lambda) cannot be retrieved due to no z variable included in the model.")`

Reply:
The message has been modified.